src/clj/ataru/hakija/hakija_application_service.clj

```diff
@@ -160,9 +164,11 @@
         applied-hakukohteet           (filter #(contains? (set (:hakukohde application)) (:oid %))
                                               hakukohteet)
         applied-hakukohderyhmat       (mapcat :hakukohderyhmat applied-hakukohteet)
-        virkailija-secret             (if is-modify?
-                                        (valid-virkailija-update-secret application)
-                                        (valid-virkailija-create-secret application))
+        rewrite-secret?               (valid-virkailija-rewrite-secret application)
+        virkailija-secret             (or rewrite-secret?
```

Comment:
Järjestäisin tän niin päin, että `valid-virkailija-rewrite-secret` testataan vain jos `is-modify?`.